testing/test_mud.sh

```diff
@@ -47,17 +47,43 @@ function test_device_traffic {
     echo device-$device_num $type $((bfr_peer > 2)) $((bfr_ngbr > 0)) $((ufr_peer > 2)) $((ufr_ngbr > 0)) | tee -a $TEST_RESULTS
 }
 
+function test_acl_count {
+    device_num=$1
+    peer_num=$((3-device_num))
+    device_mac=9a:02:57:1e:8f:0$device_num
+    peer_mac=9a:02:57:1e:8f:0$peer_num
+    metric_output_file=inst/run-9a02571e8f0$device_num/metric_output.json
+    rule_filter=eth_src=$peer_mac,eth_dst=$device_mac
+
+    python3 daq/varz_state_collector.py -y flow_packet_count_port_acl -o $metric_output_file -l $rule_filter
+    packet_count=$(jq '.gauge_metrics.flow_packet_count_port_acl.samples[0].value' $metric_output_file)
+    echo device-$device_num $type $((packet_count > 2)) | tee -a $TEST_RESULTS
+}
+
+function terminate_daq {
+    daq_pid=$(<inst/daq.pid)
+    sudo kill -SIGINT $daq_pid
+    while [ -f inst/daq.pid ]; do
+        sleep 5
+    done
+}
+
 function test_mud {
     type=$1
     echo %%%%%%%%%%%%%%%%% test mud profile $type
-    cmd/run -s device_specs=resources/device_specs/bacnet_$type.json
+    cmd/run -k -s device_specs=resources/device_specs/bacnet_$type.json & sleep 120
```

Comment:
move sleep to separate line

Reply:
also, would this be better served by network mode (-n) rather than single-shot and hold?  Do you actually need to run the tests, or just have the system setup the network?  If you're not sure what network mode is then maybe spend just a bit of time exploring what it is and then see if it applies.
Also, you can use the "timeout" command to make some of this easier to read, maybe.  E.g., "timeout 120s XXX" -- you'll still need your kill-daq script (because it doesn't kill subshells), but it doesn't involve directly running the background process.

Reply:
So the problem I had with "-n" mode is that the devices cannot get IP addresses. I guess that's because the DHCP test is not run... As a result the devices could not generate Bacnet traffic...